sklearn/linear_model/_ridge.py

```diff
@@ -1842,13 +1844,42 @@ def fit(self, X, y, sample_weight=None):
         if sample_weight is not None:
             sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
 
-        self.alphas = np.asarray(self.alphas)
-
-        if np.any(self.alphas <= 0):
-            raise ValueError(
-                "alphas must be strictly positive. Got {} containing some "
-                "negative or null value instead.".format(self.alphas)
+        if isinstance(self.alphas, (np.ndarray, list, tuple)):
+            n_alphas = 1 if np.ndim(self.alphas) == 0 else len(self.alphas)
+            if n_alphas != 1:
+                for index, alpha in enumerate(self.alphas):
+                    alpha = check_scalar(
+                        alpha,
+                        f"alphas[{index}]",
+                        target_type=numbers.Real,
+                        min_val=0.0,
+                        include_boundaries="neither",
+                    )
+            elif np.ndim(self.alphas) == 0:
+                self.alphas[()] = check_scalar(
+                    self.alphas[()],
+                    "alphas",
+                    target_type=numbers.Real,
+                    min_val=0.0,
+                    include_boundaries="neither",
+                )
+            else:
+                self.alphas[0] = check_scalar(
+                    self.alphas[0],
+                    "alphas",
+                    target_type=numbers.Real,
+                    min_val=0.0,
+                    include_boundaries="neither",
+                )
+        else:
```

Comment:
It could also be good to add a comment indicating which type of `alphas` to we expect in this branch of the code (I assume a single float).